source/verus/src/main.rs

```diff
@@ -75,19 +63,270 @@ fn main() {
+            std::fs::copy(dep, proj_path.join(&rel_dest)).map_err(|err| {
+                format!(
+                    "failed to copy file {} to repo_path with error message: {}",
+                    dep.display(),
+                    err
+                )
+            })?;
+        }
+    } else {
+        // copy files to repo_path, none of dep should be absolute path at thsi point
+        for dep in deps.iter() {
+            create_dir_all(proj_path.join(dep.parent().unwrap())).unwrap();
+            std::fs::copy(dep, proj_path.join(dep)).map_err(|err| {
+                format!(
+                    "failed to copy file {} to repo_path with error message: {}",
+                    dep.display(),
+                    err
+                )
+            })?;
         }
     }
+
+    std::fs::remove_file(&dep_file_path).map_err(|err| {
+        format!(
+            "failed to remove dependency file {} in local data cache with error message: {}",
+            dep_file_path.display(),
+            err
+        )
+    })?;
+
+    Command::new("git").current_dir(&proj_path).arg("add").arg(".").output().map_err(|x| {
+        format!("localstorage: failed to stage current project with error message: {}", x)
+    })?;
+
+    std::process::Command::new("git")
+        .current_dir(&proj_path)
+        .arg("commit")
+        .arg("-m")
+        .arg("\"verus\"")
+        .env("GIT_AUTHOR_NAME", "nobody")
+        .env("GIT_AUTHOR_EMAIL", "nobody@nobody.nobody")
+        .env("GIT_COMMITTER_NAME", "nobody")
+        .env("GIT_COMMITTER_EMAIL", "nobody@nobody.nobody")
+        .output()
+        .map_err(|x| format!("localstorage: failed to git commit with error message: {}", x))?;
+
+    Ok(())
+}
+
+#[derive(Debug)]
+pub struct Reports {
+    proj_path: PathBuf,
+    dep_path: PathBuf,
+}
+
+fn repo_path(store: bool) -> Option<Reports> {
+    // check if user has git as executable
+    if Command::new("git").arg("--version").output().is_err() {
+        return None;
+    }
+
+    // check if user has local storage enabled during vargo build
+    if option_env!("VERUS_LOCAL_STORE").is_none() {
+        return None;
+    }
+
+    // Check if there's a verus/reports/.git file in the user's local data lib
+    // if not so, create verus/ directory
+    let reports_dir = dirs::data_local_dir()?.join("verus").join("reports");
+
+    if !reports_dir.join(".git").is_file() {
+        create_dir_all(&reports_dir).expect("Creating verus/ directory in local data dir");
+
+        Command::new("git")
+            .current_dir(&reports_dir)
+            .arg("init")
+            .output()
+            .expect("Initializing git in local data dir");
+    }
+
+    let uuid_file = reports_dir.join("userid");
+    if !uuid_file.is_file() {
+        let uuid = uuid::Uuid::new_v4();
+        write(&uuid_file, uuid.to_string()).expect("Writing user id file");
+    }
+
+    let uuid = read_to_string(&uuid_file).expect("Reading user id file");
```

Comment:
Rather than calling it a `uuid` in your variables, can you call it `user_id` or such? Best to use "what is this" rather than "what format is this in".